### PR TITLE
v0.14

### PR DESCRIPTION
**Pull request by kejace**
1. support purescript v0.14
2. support halogen v6
3. use all the updated versions of `puresript-*` accordingly

I have verified that this works, building using `make parcel-start`. Note that this validates all the upstream dependencies, including `chanterelle` and `purescript-web3-generator`.

To test this, make your life easier by cloning fresh, or run what I used:

```
npm uninstall -g chanterelle && rm -rf output && npm i -g f-o-a-m/chanterelle#v0.14 && chanterelle global-postinstall && rm -rf purs/src/Contracts && make build-contracts && make build-purs
```

This helps you out of the dependency loop that `chanterelle` creates.

Note that the test fails for me locally with

```
2021-07-09T19:41:14.099Z [INFO] Contract SimpleStorage deployed to address 0x2ffbf3db2a0f928ae71f1f22f421ee1a69631bcb
/home/kejace/devel/chanterelle-halogen-template/output/Effect.Aff/foreign.js:532
                throw util.fromLeft(step);
                ^

TypeError: eff is not a function
    at runSync (/home/kejace/devel/chanterelle-halogen-template/output/Effect.Aff/foreign.js:90:20)
    at run (/home/kejace/devel/chanterelle-halogen-template/output/Effect.Aff/foreign.js:328:22)
    at /home/kejace/devel/chanterelle-halogen-template/output/Effect.Aff/foreign.js:348:19
    at drain (/home/kejace/devel/chanterelle-halogen-template/output/Effect.Aff/foreign.js:120:9)
    at Object.enqueue (/home/kejace/devel/chanterelle-halogen-template/output/Effect.Aff/foreign.js:141:11)
    at /home/kejace/devel/chanterelle-halogen-template/output/Effect.Aff/foreign.js:339:27
    at /home/kejace/devel/chanterelle-halogen-template/output/Effect.Aff.Compat/index.js:18:55
    at /home/kejace/devel/chanterelle-halogen-template/output/Network.Ethereum.Web3.JsonRPC/foreign.js:11:21
    at XMLHttpRequest.request.onreadystatechange (/home/kejace/devel/chanterelle-halogen-template/node_modules/ethjs-provider-http/lib/index.js:91:7)
    at XMLHttpRequestEventTarget.dispatchEvent (/home/kejace/devel/chanterelle-halogen-template/node_modules/xhr2/lib/xhr2.js:64:18)
[error] Tests failed: exit code: 1
Makefile:39: recipe for target 'test-purs-dapp' failed
make: *** [test-purs-dapp] Error 1
```

The following posts seem relevant, and this is likely actually an issue in `purescript-web3` since that's where `JSONRpc.js` comes from.

1. https://discourse.purescript.org/t/halogen-typeerror-eff-is-not-a-function/2008
2. https://github.com/purescript/documentation/blob/4f6c0e3b8b6184329ab1e9f7ec3c0f5a5e27a393/guides/FFI-Tips.md#why-doesnt-my-effect-work-when-passed-to-a-normal-js-function

### CODE REVIEW

**Comment by kejace** [comment]
@iostat I'm getting a CI error relating to `chanterelle` (note that CI on corresponding PR for `chanterelle` works)

```
npm ERR! code ENOENT
npm ERR! syscall chmod
npm ERR! path /home/travis/.nvm/versions/node/v16.4.2/lib/node_modules/chanterelle/chanterelle-bin.sh
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, chmod '/home/travis/.nvm/versions/node/v16.4.2/lib/node_modules/chanterelle/chanterelle-bin.sh'
```

Could you attempt to fix this?

**Comment by kejace** [comment]
`11` is the magic node version

File: purs/src/App/Component/Image.purs [hunk 1]
```diff
@@ -15,13 +14,15 @@ import Halogen.HTML.Properties as HP
   still loading, we indicate this. If the image failed to load, we increment a retry count
   and try again.
 -}
+data Query :: forall k. k -> Type
```
Comment (martyall) [inline purs/src/App/Component/Image.purs]: is this required?

File: purs/src/App/Component/Image.purs [hunk 2]
```diff
@@ -15,13 +14,15 @@ import Halogen.HTML.Properties as HP
...
 data Query a
 
 data Action
   = LoadFailed
   | LoadSucceeded
   | RetryLoading
 
+type Slots :: forall k. Row k
```
Comment (martyall) [inline purs/src/App/Component/Image.purs:25]: is this required?

**Review by martyall — CHANGES_REQUESTED** [review]

File: purs/src/App/Component/Image.purs (hunk 2 above)
Comment (kejace) [reply purs/src/App/Component/Image.purs:25]: I'm not able to get rid of this constraint, without getting a warning.

**Review by martyall — APPROVED** [review]

**Pull request merged**